src/lib/wpcli-versions.ts

```diff
@@ -0,0 +1,78 @@
+import fs from 'fs-extra';
+import semver from 'semver';
+import { downloadWPCLI } from '../../vendor/wp-now/src/download';
+import { executeWPCli } from '../../vendor/wp-now/src/execute-wp-cli';
+import getWpCliPath from '../../vendor/wp-now/src/get-wp-cli-path';
+
+export async function updateLatestWPCliVersion() {
+	let shouldOverwrite = false;
+	const pathExist = await fs.pathExists( getWpCliPath() );
+	if ( pathExist ) {
+		shouldOverwrite = await isWPCliInstallationOutdated();
+	}
+	await downloadWPCLI( shouldOverwrite );
+}
+
+export async function isWPCliInstallationOutdated(): Promise< boolean > {
+	const installedVersion = await getWPCliVersionFromInstallation();
+	const latestVersion = await getLatestWPCliVersion();
+
+	if ( ! installedVersion ) {
+		return true;
+	}
+
+	if ( ! latestVersion ) {
+		return false;
+	}
+
+	console.log( 'Installed WP-CLI version:', installedVersion );
+	console.log( 'Latest WP-CLI version:', latestVersion );
+	console.log( 'Is WP-CLI outdated:', semver.lt( installedVersion, latestVersion ) );
+	try {
+		return semver.lt( installedVersion, latestVersion );
+	} catch ( _error ) {
+		return false;
+	}
+}
+
+let latestWPCliVersionCache: string | null = null;
+
+async function getLatestWPCliVersion() {
+	// Only fetch the latest version once per app session
+	if ( latestWPCliVersionCache ) {
+		return latestWPCliVersionCache;
+	}
+
+	try {
+		const response = await fetch(
+			'https://api.github.com/repos/wp-cli/wp-cli/releases?per_page=1'
+		);
+		const data: Record< string, string >[] = await response.json();
+		latestWPCliVersionCache = data?.[ 0 ]?.tag_name || '';
+	} catch ( _error ) {
+		// Discard the failed fetch, return the cache
+	}
+
+	return latestWPCliVersionCache || '';
+}
+
+const getWPCliVersionFromInstallation = async () => {
+	const originalConsoleLog = console.log;
+	const logMessages: string[] = [];
+
+	console.log = function ( args ) {
```

Comment:
This seems to be a bit hacky, and it results in printing the output. Is there another way to call WP CLI and get the output?

Reply:
I agree, but unfortunatelly, no. WP-CLI has another command named `wp cli version` but that doesn't return any results as well. You can take a look [here](https://github.com/wp-cli/wp-cli/blob/main/features/cli.feature) to see that this is the case according to tests.
To my understanding, this is the way to go unfortunatelly. :disappointed: 

@sejas , I know you are familiar with `wp-cli`. Are is my assumption correct? What do you think?

Reply:
We will add wp-cli inline execution here: https://github.com/Automattic/dotcom-forge/issues/7328
We could wait until that's done before proceeding with this approach. You can assume `executeWPCli` will return the output `result.text` as a string, similar to `php.run`.

Reply:
I've created a PR https://github.com/Automattic/studio/pull/222 to receive the output from the function. Feel free to merge it.